src/components/Auth/AuthHeader.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { FC, memo, useCallback, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import { useCurrentCriteriaContext } from 'components/CurrentCriteriaProvider';
+import SignupModal from './SignupModal';
+import LoginModal from './LoginModal';
+import ResetModal from './ResetModal';
+import LinkSentModal from './LinkSentModal';
+import NewPasswordModal from './NewPasswordModal';
+import PasswordResetModal from './PasswordResetModal';
+
+import logo from 'camino-builder-logo-light-bg.png';
+import useStyles from './styles';
+
+const AuthHeader: FC = () => {
+  const styles = useStyles();
+
+  const [openLogin, setOpenLogin] = useState<boolean>(false);
```

Comment:
Instead of defining like 6 different modals can it not just be one modal which renders differently based on the props?